testutils/check-calling-form-in-baseline/main.go

```diff
@@ -0,0 +1,218 @@
+// Copyright 2018 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"bufio"
+	"flag"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"log"
+	"os"
+	"path/filepath"
+	"regexp"
+	"sort"
+	"strings"
+)
+
+const usage = `Reads calling forms from gapic_yaml_file and makes sure each is used by
+some *.baseline file in dir.
+The line specifying the calling form is in the format:
+
+  callingFormCheck: <valueSetID> <language>: <callingForm1> <callingForm2> ...
+
+The string "callingFormCheck" must start the line, preceded only by a possilby empty
+run of whitespaces. To check for multiple languages or vaule sets, simply write the line
+multiple times. Hint: YAML files allow multi-line string literals like
+
+  sometext: >
+    callingFormCheck: foo bar: zip zap
+    callingFormCheck: spam eggs: ham
+`
+
+func main() {
+	yamlFname := flag.String("yaml", "", "gapic yaml file")
+	flag.Usage = func() {
+		binName := os.Args[0]
+		out := os.Stderr // Should be flag.CommandLine.Output(), but Go version on Travis is too old.
+		fmt.Fprintf(out, "Usage of %s:\n", binName)
+		fmt.Fprintf(out, "%s -yaml <gapic_yaml_file> [dir]\n", binName)
+		flag.PrintDefaults()
+		fmt.Fprint(out, usage)
+	}
+	flag.Parse()
+
+	if *yamlFname == "" {
+		flag.Usage()
+		os.Exit(1)
+	}
+	yamlf, err := os.Open(*yamlFname)
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer yamlf.Close()
+
+	checks, err := readChecks(yamlf)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	walkFn := func(path string, info os.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
+		if len(checks) == 0 {
+			return filepath.SkipDir
+		}
+		if !info.Mode().IsRegular() {
+			return nil
+		}
+		if filepath.Ext(path) != ".baseline" {
+			return nil
+		}
+		baseline, err := ioutil.ReadFile(path)
+		if err != nil {
+			return err
+		}
+		lang := filepath.Base(filepath.Dir(path))
+		delFoundForms(string(baseline), lang, checks)
+		return nil
+	}
+	dir := "."
+	if flag.NArg() > 0 {
+		dir = flag.Arg(0)
+	}
+	if err := filepath.Walk(dir, walkFn); err != nil {
+		log.Fatal(err)
+	}
+
+	if len(checks) > 0 {
+		var errs []checkConfig
+		for c := range checks {
+			errs = append(errs, c)
+		}
+		sort.Sort(checkConfigSlice(errs)) // Not sort.Slice, Go version on Travis is too old.
+		fmt.Println("not found:")
```

Comment:
Suggest adding two more data that will be helpful in case of failure:

"The following value set/calling form combinations specified in the $yaml file were not found in the $baseline file:"

Reply:
Done